pkg/daemon/agent/flexvolume/server.go

```diff
@@ -106,30 +109,47 @@ func (s *FlexvolumeServer) Start() error {
 }
 
 // Stop the unix domain socket server and deletes the socket file
-func (s *FlexvolumeServer) Stop() {
-	if s.listener != nil {
-		logger.Info("Stopping unix socket rpc server.")
-		if err := s.listener.Close(); err != nil {
+func (s *FlexvolumeServer) StopAll() {
+	logger.Infof("Stopping %d unix socket rpc server(s).", len(s.listeners))
+	for unixSocketFile, listener := range s.listeners {
+		if err := listener.Close(); err != nil {
 			logger.Errorf("Failed to stop unix socket rpc server: %+v", err)
 		}
+
+		// closing the listener should remove the unix socket file. But lets try it remove it just in case.
+		if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
+			logger.Infof("Deleting unix domain socket file %s.", unixSocketFile)
+			os.Remove(unixSocketFile)
+		}
 	}
-	// closing the listener should remove the unix socket file. But lets try it remove it just in case.
-	unixSocketFile := path.Join(flexVolumeDriverDir, path.Join(UnixSocketName)) // /flextmnt/rook.io~rook/.rook.sock
-	if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
-		logger.Info("Deleting unix domain socket file.")
-		os.Remove(unixSocketFile)
-	}
+	s.listeners = make(map[string]net.Listener)
```

Comment:
this map gets cleared even for listeners that failed to close.  is that OK?

Reply:
In the production code path, `StopAll()` is only called when the process is terminating, so even clearing the map is technically unnecessary